typing/out_type.ml

```diff
@@ -1054,6 +1080,18 @@ let rec tree_of_typexp mode ty =
             | _ -> Otyp_stuff "<hidden>"
           else tree_of_typexp mode ty1 in
         Otyp_arrow (lab, t1, tree_of_typexp mode ty2)
+    | Tfunctor (l, id, (p, fl), ty) ->
+        let lab =
+          if !print_labels || is_optional l then l else Nolabel
+        in
+        let fenv env =
+          let mty = !Ctype.modtype_of_package env Location.none p fl in
+          Env.add_module ~arg:true (Ident.of_unscoped id) Mp_present mty env
+        in
+        let ty = wrap_env fenv (tree_of_typexp mode) ty in
```

Comment:
A note, maybe more for @Octachron: as we only extend the current environment with a new module (no new equalities), the previous shortening choices remain valid. I wonder if we could have a variant of `wrap_env` (or a new parameter to say that) that makes a copy of the shortening structures, instead of reseting them to their initial state.